apps/storefront/src/pages/QuoteDraft/index.tsx

```diff
@@ -498,6 +503,47 @@ function QuoteDraft({ setOpenPage }: PageProps) {
     }
   };
 
+  /**
+   * Clone address and compare with selectedAddress to decide if addressId can be reused.
+   *
+   * @param address - Address with optional selectedAddress for comparison
+   * @returns Cloned address, potentially with reused addressId if no changes detected
+   *
+   * Cases:
+   * 1. No Selected Address: return cloned address
+   * 2. Selected Address copy exists and address unchanged: return cloned address with addressId
+   * 3. Selected Address copy exists but address changed: return cloned address without addressId
+   */
+  const cloneAddressWithId = (address: AddressWithSelected): ShippingAddress | BillingAddress => {
```

Comment:
```suggestion
  const cloneAddressWithId = ({ selectedAddress, ...address }: AddressWithSelected): ShippingAddress | BillingAddress => {
```

seems like something like this would have been easier

Reply:
resolved in this [PR](https://github.com/bigcommerce/b2b-buyer-portal/pull/583)